pkg/controller/cronjob_trigger_controller.go

```diff
@@ -0,0 +1,358 @@
+		return
+	}
+
+	c.logger.Info("Cron Job Trigger controller synced and ready")
+
+	wait.Until(c.runWorker, time.Second, stopCh)
+}
+
+// WaitForCacheSync is required for caches to be synced
+func (c *CronJobTriggerController) WaitForCacheSync(stopCh <-chan struct{}) bool {
+	if !cache.WaitForCacheSync(stopCh, c.cronJobInformer.Informer().HasSynced, c.functionInformer.Informer().HasSynced) {
+		utilruntime.HandleError(fmt.Errorf("Timed out waiting for caches required for Cronjob triggers controller to sync;"))
+		return false
+	}
+	c.logger.Info("Cronjob Trigger controller caches are synced and ready")
+	return true
+}
+
+func (c *CronJobTriggerController) runWorker() {
+	for c.processNextItem() {
+		// continue looping
+	}
+}
+
+func (c *CronJobTriggerController) processNextItem() bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.syncCronJobTrigger(key.(string))
+	if err == nil {
+		// No error, reset the ratelimit counters
+		c.queue.Forget(key)
+	} else if c.queue.NumRequeues(key) < cronJobTriggerMaxRetries {
+		c.logger.Errorf("Error processing %s (will retry): %v", key, err)
+		c.queue.AddRateLimited(key)
+	} else {
+		// err != nil and too many retries
+		c.logger.Errorf("Error processing %s (giving up): %v", key, err)
+		c.queue.Forget(key)
+		utilruntime.HandleError(err)
+	}
+
+	return true
+}
+
+func (c *CronJobTriggerController) syncCronJobTrigger(key string) error {
+	c.logger.Infof("Processing update to CronJob Trigger: %s", key)
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	obj, exists, err := c.cronJobInformer.Informer().GetIndexer().GetByKey(key)
+	if err != nil {
+		return fmt.Errorf("Error fetching object with key %s from store: %v", key, err)
+	}
+
+	// this is an update when CronJob trigger API object is actually deleted, we dont need to process anything here
+	if !exists {
+		c.logger.Infof("Cronjob Trigger %s not found, ignoring", key)
+		return nil
+	}
+
+	cronJobtriggerObj := obj.(*cronjobTriggerAPi.CronJobTrigger)
+
+	// CronJob trigger API object is marked for deletion (DeletionTimestamp != nil), so lets process the delete update
+	if cronJobtriggerObj.ObjectMeta.DeletionTimestamp != nil {
+
+		// If finalizer is removed, then we already processed the delete update, so just return
+		if !c.cronJobTriggerObjHasFinalizer(cronJobtriggerObj) {
+			return nil
+		}
+
+		// CronJob Trigger object should be deleted, so remove associated cronjob and remove the finalizer
+		err = c.clientset.BatchV1beta1().CronJobs(ns).Delete(name, &metav1.DeleteOptions{})
+		if err != nil && !k8sErrors.IsNotFound(err) {
+			c.logger.Errorf("Failed to remove CronJob created for CronJobTrigger Obj: %s due to: %v: ", key, err)
+			return err
+		}
+
+		// remove finalizer from the cronjob trigger object, so that we dont have to process any further and object can be deleted
+		err = c.cronJobTriggerObjRemoveFinalizer(cronJobtriggerObj)
```

Comment:
AFAIK we don't know a finalizer here. The cronjob has the same name than the trigger so we don't need to preserve the trigger object until we delete it.